lib/git_repository.rb

```diff
@@ -91,6 +91,21 @@ def self.directory(path)
     path.split("/")[0..-2].join("/")
   end
 
+  def deepest_existing_dir(path, commit_oid = nil)
+    path ||= '/'
+    dirs = path.split('/')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.